builder/builder.go

```diff
@@ -379,14 +396,23 @@ func (b *Builder) rootOwnedDir(path string, time time.Time) *tar.Header {
 }
 
 func (b *Builder) orderLayer(dest string) (string, error) {
-	orderTOML := &bytes.Buffer{}
-	err := toml.NewEncoder(orderTOML).Encode(OrderTOML{Groups: b.metadata.Groups})
+	buf := &bytes.Buffer{}
+	lifecycleVersion := b.GetLifecycleVersion()
+
+	var tomlData interface{}
+	if lifecycleVersion != nil && lifecycleVersion.LessThan(semver.MustParse("0.4.0")) {
+		tomlData = v1OrderTOMLFromOrderTOML(orderTOML{Order: b.metadata.Groups.ToConfig()})
```

Comment:
why not just make one directly from an Order instead of creating an intermediate `orderTOML`?